[DPE-6961] Add how-to documentation for AKS

theoctober19th opened the pull request:
This PR is only for getting review feedbacks on the documentation. I'll update the contents in Discourse once it gets approved.

theoctober19th committed:
Add documentation for AKS

theoctober19th committed:
Polish doc

theoctober19th committed:
Improve documentation

theoctober19th commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
````diff
@@ -149,6 +149,104 @@ users:
+Once terraform is installed, we're ready to write the Terraform plan for our new AKS cluster. The terraform script for the creation of a simple AKS cluster for Charmed Spark is already available [here](https://github.com/theoctober19th/charmed-spark-aks). Let's clone the repo and inspect a little bit into the content. 
+
+```bash
+git clone https://github.com/theoctober19th/charmed-spark-aks.git
````
I'm not sure whether or not we should put this code in Canonical's namespace. Any feedback here is appreciated.

Batalex replied:
IIRC one item for next cycle is providing architecture references with TF. This will probably be one of them, so yes, I would say it will go into Canonical's namespace at some point. Not sure how or where, though.

We could copy the content of the repo into the `resources` folder for now

deusebio replied:
Yes, we should definitely have this in the canonical repo. It is also fine to place the code here or place it in the resource folder as @Batalex suggested. 

Once that we will have the proper reference architecture, we can then refer to that one and refactor this part

theoctober19th replied:
Thank you guys for the suggestions, I've moved the TF module to `spark-k8s-bundle` repo itself. Here is another PR for the migration: https://github.com/canonical/spark-k8s-bundle/pull/102

Batalex commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -149,6 +149,104 @@ users:
+
+Browse to the page https://microsoft.com/devicelogin, enter the code displayed in the console earlier, and finally login with your Azure account credentials. When prompted for confirmation, click "Yes". Once the authentication is successful, you should see authentication success message in the browser. If you have multiple subscriptions associated with your Azure account, you maybe asked to choos ...
+
+To cofirm that the authentication went well, you can try listing storage accounts in your Azure Cloud:
```
```suggestion
To confirm that the authentication went well, you can try listing storage accounts in your Azure Cloud:
```

Batalex reviewed (commented):
Nice job with the authentication flow, it will be helpful for everyone

deusebio reviewed (commented):
Looks good! Just place the code in the bundle repository and reference it in the docs (I provide below a suggestion) and I believe we should be good to go!

theoctober19th committed:
Update how-to guide

theoctober19th committed:
Update releases/3.4/yaml/docs/how-to/h-setup-k8s.md

Co-authored-by: Alex Batisse <alex.batisse@canonical.com>

Batalex commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -199,17 +199,23 @@ Terraform is an infrastructure as code (IaC) tool that lets you create, change a
-Once terraform is installed, we're ready to write the Terraform plan for our new AKS cluster. The terraform script for the creation of a simple AKS cluster for Charmed Spark is already available [here](https://github.com/theoctober19th/charmed-spark-aks). Let's clone the repo and inspect a little bit into the content. 
+Once terraform is installed, we're ready to write the Terraform plan for our new AKS cluster. 
+
+The terraform scripts for the creation of a simple AKS cluster for Charmed Spark is already available [here](https://github.com/canonical/spark-k8s-bundle/tree/add-aks-setup/releases/3.4/yaml/docs-resources/aks-setup). If you inspect the contents there, you will see that there are different Terraform configuration files. These configuration files contain the specification for the resource group,  ...
```
**todo**: let's use a relative path instead of an absolute one that uses the branch name

theoctober19th replied:
Of course, I've not changed it yet since the other PR that merges it to main has not been merged. This will be updated on the Discourse.

theoctober19th replied:
>  let's use a relative path instead of an absolute one that uses the branch name

WDYM by relative path? Since this will be published in Discourse, we would need full URL to the TF module there.

Batalex commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -199,17 +199,23 @@ Terraform is an infrastructure as code (IaC) tool that lets you create, change a
+# file: setup.tf
+
+module "aks_cluster" {
+  source     = "git::https://github.com/canonical/spark-k8s-bundle//releases/3.4/yaml/docs-resources/aks-setup?ref=add-aks-setup"
```
**todo**: Same here, let's remove the rev path fragment because we will have a dead link

theoctober19th replied:
Of course, I've not changed it yet since the other PR that merges it to main has not been merged. This will be updated on the Discourse.

theoctober19th replied:
I've updated the links, nevertheless -- just so it doesn't get missed when I copy the contents to Discourse.

theoctober19th committed:
Use the TF bundle from main branch

theoctober19th committed:
Merge branch 'eks-doc' of github.com:canonical/spark-k8s-bundle into eks-doc

izmalk commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -149,6 +149,110 @@ users:
+
+#### Azure AKS
+
+To deploy an Azure Kubernetes Service (AKS) cluster, you'd need to make sure you have Azure CLI properly installed and authenticated on your edge machine.
```
Suggestion: Terminology: do we want to call the local machine "edge"?

```suggestion
To deploy an Azure Kubernetes Service (AKS) cluster, you'd need to make sure you have Azure CLI properly installed and authenticated on your local machine.
```

theoctober19th replied:
This can either be a local machine or a remote VM as well.

izmalk replied:
I wonder what is the correct terminology that we can use. What kind of machine that can be. "Your machine"? Alternatively, we can delete the reference to machine entirely:
```suggestion
To deploy an Azure Kubernetes Service (AKS) cluster, you'd need to make sure you have Azure CLI properly installed and authenticated.
```

theoctober19th replied:
Okay, let's delete the reference to machine altogether.

izmalk committed:
Wording improvements

Wording adjusted to be more imperative - it's more suitable for a HowTo guide this way. All the "let's" and "can" are usually reserved for tutorials.
Link text changed to not have "here".
Added a link to Terraform in its introduction.
Line splits - splitting longer lines to limit the scope of changes in git commits later.
Removed redundant empty lines.
Other minor improvements.

izmalk committed:
Minor typo fix

izmalk commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -149,6 +149,117 @@ users:
+# file: setup.tf
+
+module "aks_cluster" {
+  source     = "git::https://github.com/canonical/spark-k8s-bundle//releases/3.4/yaml/docs-resources/aks-setup"
```
Question: Are you sure the link is correct? Especially the `git::http::` part.
```suggestion
  source     = "git::https://github.com/canonical/spark-k8s-bundle//releases/3.4/yaml/docs-resources/aks-setup"
```

theoctober19th replied:
Yes, the link is correct and I've verified this as well.

izmalk commented on releases/3.4/yaml/docs/how-to/h-setup-k8s.md:
```diff
@@ -149,6 +149,117 @@ users:
+
+Once terraform is installed, we're ready to write the Terraform plan for our new AKS cluster.
+
+The terraform scripts for the creation of a simple AKS cluster for Charmed Spark is already available in the [GitHub repository](https://github.com/canonical/spark-k8s-bundle/tree/main/releases/3.4/yaml/docs-resources/aks-setup).
```
Question: Can we add the folder in this PR? Even if it won't be merged here.

theoctober19th replied:
I can pull that commit here -- however since we're going to merge that through the other PR anyway, it would just be redundant. You can view the folder structure here: https://github.com/canonical/spark-k8s-bundle/tree/add-aks-setup/releases/3.4/yaml/docs-resources/aks-setup

izmalk reviewed (approved):
I've committed a few wording improvements and added a few questions to check. Nothing blocking. LGTM.

izmalk committed:
Second minor typo fix

theoctober19th committed:
Update releases/3.4/yaml/docs/how-to/h-setup-k8s.md

Co-authored-by: Vladimir Izmalkov <48120135+izmalk@users.noreply.github.com>

theoctober19th commented:
I've updated the docs directly in Discourse. Thank you everyone for your valuable reviews and input. Closing this PR now.
